bucket4j/src/main/java/com/linecorp/armeria/server/throttling/tokenbucket/TokenBucket.java

```diff
@@ -0,0 +1,140 @@
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.throttling.tokenbucket;
+
+import static java.util.Objects.requireNonNull;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.google.common.base.MoreObjects;
+
+/**
+ * Stores configuration of the Token-Bucket algorithm, comprised of multiple limits.
+ */
+public class TokenBucket {
+
+    @Nonnull
+    private final BandwidthLimit[] limits;
+
+    /**
+     * Defines throttling configuration comprised of zero or more bandwidth limits in accordance to
+     * token-bucket algorithm.
+     *
+     * <h3>Multiple bandwidths:</h3>
+     * It is possible to specify more than one bandwidth per bucket, and bucket will handle all bandwidth in
+     * strongly atomic way. Strongly atomic means that token will be consumed from all bandwidth or from
+     * nothing, in other words any token can not be partially consumed.
+     * <br> Example of multiple bandwidth:
+     * <pre>{@code
+     * // Adds bandwidth that restricts to consume
+     * // not often than 1000 tokens per 1 minute and
+     * // not often than 100 tokens per second.
+     * TokenBucketConfig config = TokenBucketConfig.builder()
+     *      .limit(1000L, Duration.ofMinutes(1))
+     *      .limit(100L, Duration.ofSeconds(1))
+     *      .build()
+     * }</pre>
+     *
+     * @param limits one or more bandwidth limits to be used by token-bucket algorithm
+     */
+    TokenBucket(@Nonnull BandwidthLimit... limits) {
+        this.limits = requireNonNull(limits, "limits");
+    }
+
+    /**
+     * Creates a new {@link TokenBucketBuilder} using a builder.
+     */
+    public static TokenBucketBuilder builder() {
+        return new TokenBucketBuilder();
+    }
+
+    /**
+     * Creates a new {@link TokenBucket} from a comma-separated {@code specification} string
+     * that conforms to the following format,
+     * as per <a href="https://tools.ietf.org/id/draft-polli-ratelimit-headers-00.html">RateLimit Header Scheme for HTTP</a>:
+     * <pre>{@code
+     * <bandwidth limit 1>[, <bandwidth limit 2>[, etc.]]
+     * }</pre>
+     * The order of elements inside {@code specification} is not defined.
+     * For example:
+     * <ul>
+     *   <li>{@code 100;window=60;burst=1000, 50000;window=3600}</li>
+     * </ul>
+     *
+     * @param specification the specification used to create a {@link BandwidthLimit}
+     * @see TokenBucketSpec#parseTokenBucket(String)
+     */
+    public static TokenBucket of(String specification) {
+        return TokenBucketSpec.parseTokenBucket(specification);
+    }
```

Comment:
Question: Would it make sense to add `of(long limit)` as well?

Reply:
Do you mean `of(BandwidthLimit limit)`?

Reply:
I meant `of(long limit)` where other properties are set with the default values. I'm not sure it will be useful to a user though. Let me know what you think.

Reply:
Minimum required set of parameters includes both `limit` and `duration`. In that case it could be `of(long limit, Duration period)`.
Added this to support basic cases.